.github/workflows/pr-opened.yml

```
@@ -0,0 +1,27 @@
+    needs: [docker-check]
+    if: ${{ needs.docker-check.outputs.docker_changed == 'true' }}
+    uses: ./.github/workflows/docker-build.yml
+
+  # Run the main tests
+  test:
+    needs: [docker-build]
+    # Allow skipping docker-build if no Docker files changed
+    if: always() && !failure() && !cancelled()
+    uses: ./.github/workflows/pipeline.yml
+    with:
+      container_tag: ${{ needs.docker-check.outputs.docker_changed == 'true' && format('pr-{0}', github.event.pull_request.number) || 'latest' }}
```

actionlint (1.7.4)

property "docker-check" is not defined in object type {docker-build: {outputs: {}; result: string}}

(expression)